infra/fastly/terraform/dl-sandbox.k8s.dev/vcl/gcs-auth.vcl

```diff
@@ -0,0 +1,76 @@
+# VCL snippet to authenticate Fastly requests to GCS.
+#
+# https://developer.fastly.com/solutions/examples/google-cloud-storage-origin-private/
+
+sub set_google_auth_header {
+
+declare local var.googleAccessKey STRING;
+declare local var.googleSecretKey STRING;
+declare local var.googleBucket STRING;
+declare local var.googleRegion STRING;
+declare local var.canonicalHeaders STRING;
+declare local var.signedHeaders STRING;
+declare local var.canonicalRequest STRING;
+declare local var.canonicalQuery STRING;
+declare local var.stringToSign STRING;
+declare local var.dateStamp STRING;
+declare local var.signature STRING;
+declare local var.scope STRING;
+
+#As of 8/2019, GCS now supports HMAC keys based on service accounts.
+#The below accessKey and secretKey should come from a service account based HMAC Key
+set var.googleAccessKey = "${access_key}";
+set var.googleSecretKey = "${secret_key}";
+set var.googleBucket = "${backend_bucket}";
+set var.googleRegion = "${region}";
+
+set bereq.http.x-amz-content-sha256 = digest.hash_sha256("");
+set bereq.http.x-amz-date = strftime({"%Y%m%dT%H%M%SZ"}, now);
+set bereq.http.host = var.googleBucket ".storage.googleapis.com";
```

Comment:
@BenTheElder I think this should solve https://github.com/kubernetes/k8s.io/issues/5727 ?

Reply:
yes! :thankyou: